File: 0387-first-unique-character-in-a-string/0387-first-unique-character-in-a-string.cpp
class Solution {
public:
    int firstUniqChar(string s) {
        int a=s.length();
        unordered_map<char,int>mp;
        for(int c:s){
            mp[c]++;
        }
        for(int i=0;i<a;i++){
            if(mp[s[i]]==1){
                return i;
            }
        }
        return -1;

    }
};

File: 0387-first-unique-character-in-a-string/0387-first-unique-character-in-a-string.java
class Solution {
    public int firstUniqChar(String s) {
        int n=s.length();
        HashMap<Character,Integer> mp = new HashMap<>();
        for(char c:s.toCharArray()){
            mp.put(c,mp.getOrDefault(c,0)+1);
        }  
        for(int i=0;i<n;i++){
            if(mp.get(s.charAt(i))==1){
                return i;
            }
        }      
        return -1;
    }
}